### ISSUE TITLE
Stupid question (new to irc in general)

### ISSUE THREAD

**Issue by austin1237** [fluffle/goirc#93]
Once connected and in a channel How exactly do I listen/see everyones message in the current channel?

**Comment by fluffle** [comment]
No question is stupid! 

**TL;DR:** Assuming you're using the example [client.go](https://github.com/fluffle/goirc/blob/master/client.go), you want to do something like this.

```go
func printMessage(conn *irc.Conn, line *irc.Line) {
    // Will print e.g.: "#go-nuts: <fluffle> I am saying something"
    fmt.Printf("%s: <%s> %s\n", line.Target(), line.Nick, line.Text())
}

func printAction(conn *irc.Conn, line *irc.Line) {
    // Will print e.g.: "#go-nuts: * fluffle does something"
    fmt.Printf("%s: * %s %s\n", line.Target(), line.Nick, line.Text())
}

func main() {
    // Put these two lines near the other two HandleFunc calls
    c.HandleFunc(irc.PRIVMSG, printMessage)
    c.HandleFunc(irc.ACTION, printAction)
}
```

You need to handle both PRIVMSG and ACTION because of the way actions (i.e. _/me does something_) are communicated in the IRC protocol. This will print out messages and actions sent to every channel that your client has JOINed. 

&lt;soapbox&gt;
You're going to need to at least skim the IRC client RFC to write anything serious with goirc, I think. A good place to start is http://ircv3.net/irc/ which in turn links to http://tools.ietf.org/html/rfc1459, the original IRC RFC. https://tools.ietf.org/html/rfc2812 updates that but the differences are minimal and probably not very relevant to you. The client does assume a reasonable working knowledge of the underlying protocol, I'm afraid.
&lt;/soapbox&gt;

I recommend trawling github for projects that use this library and looking at how they've organised their code too. For example, here's [sp0rkle](https://github.com/fluffle/sp0rkle/), an IRC bot I wrote a while ago to replace an ancient perl-fu bot. I don't recommend installing it, because the dependency on MongoDB is not very friendly for the casual user, but looking through the stuff in [main.go](https://github.com/fluffle/sp0rkle/blob/master/main.go) and [bot/*](https://github.com/fluffle/sp0rkle/tree/master/bot) could be helpful. The imaginatively named [goircbot](https://github.com/StalkR/goircbot) is by someone I work with, and does things a little differently. I'm sure there are many other examples to choose from!

I'll leave this open in case you have any further questions. Close it when you feel comfortable doing so!

**Comment by austin1237** [comment]
Luckily my use case is fairly simply. Only need to keep check if the average message count has increased as well as listen for a certain keywords in the messages. Thanks for your help. I'll close the issue
